scripts/rescript_format.js

```diff
@@ -95,26 +134,37 @@ function main(argv, rescript_exe, bsc_exe) {
         process.exit(2);
       }
       files = output.stdout.split("\n").map(x => x.trim());
-      var hasError = false;
-      for (let arg of files) {
-        if (isSupportedFile(arg)) {
-          // console.log(`processing ${arg}`);
-          child_process.execFile(
-            bsc_exe,
-            ["-o", arg, "-format", arg],
-            (error, _stdout, stderr) => {
-              if (error !== null) {
-                console.error(stderr);
-                hasError = true;
+      var incorrectlyFormattedFiles = 0;
+      try {
+        const _promises = await Promise.all(
```

Comment:
I noticed - and this applies to the old as well as the new implementation - that `bsc -format` is run in parallel for all files.

Not sure how far this will scale. Maybe this should be changed to processing in batches, but that's probably out of the scope of this PR?

Reply:
I'd say it's indeed out of scope of this PR, especially if we don't have any report of issues caused by this.